transports/transport-rest/transport-rest-client/src/main/java/org/apache/servicecomb/transport/rest/client/RestTransportClient.java

```diff
@@ -73,8 +89,21 @@ private static HttpClientOptions createHttpClientOptions() {
   }
 
   public void send(Invocation invocation, AsyncResponse asyncResp) {
-    HttpClientWithContext httpClientWithContext = findHttpClientPool(invocation);
+    URIEndpointObject endpoint = (URIEndpointObject) invocation.getEndpoint().getAddress();
+    HttpClientWithContext httpClientWithContext;
+
+    if (endpoint.isHttp2Enabled()) {
+      httpClientWithContext = findHttp2ClientPool(invocation);
+
+      if (!endpoint.isSslEnabled()) {
+        ((HttpClientImpl) httpClientWithContext.getHttpClient()).getOptions().setHttp2ClearTextUpgrade(false);
```

Comment:
endpoint affect client
are you sure it's ok?

i had wrote this review message many days ago, it's disappeared?

Reply:
one httpClient can connect to many endpoint, it's a connection pool
you change httpClient by one endpoint, then how can others work?

Reply:
Fixed and committed